iothub_client/tests/common_longhaul/iothub_client_common_longhaul.c

```diff
@@ -614,6 +621,34 @@ static int run_on_loop(RUN_ON_LOOP_ACTION action, size_t iterationDurationInSeco
     return result;
 }
 
+static void set_test_execution_and_duration(IOTHUB_LONGHAUL_RESOURCES* longhaulResources)
+{
+    char* test_total_duration_string = getenv(IOTHUB_LONGHAUL_TOTAL_DURATION_SECS);
+    char* test_loop_duration_string = getenv(IOTHUB_LONGHAUL_LOOP_DURATION_SECS);
```

Comment:
On certain C99 implementations this call may corrupt the one above. If more than one call is necessary you may want to consider allocating another string and copying the result.
In practice, AFAIK this will work fine on Linux x64 and Windows (they are pointers in the process control block's env) but I would recommend defining them as `const char*`